app/helpers/picture_version_helper.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module PictureVersionHelper
+  ##
+  # Looks up an Alchemy::PictureVersion from the DB using the
+  # from_cache method. If the image does not exist, it will be created and persisted.
+  #
+  # @param picture [Alchemy::Picture]
+  # @param settings [Hash]
+  # @return Alchemy::PictureVersion
+  def picture_version(picture, settings = {})
+    options = []
+
+    if settings[:format] && settings[:format] == 'jpeg' && settings[:quality]
+      options << "-quality #{settings[:quality]}"
+    end
+
+    unless settings[:format]
+      settings[:format] = 'jpeg'
+    end
+
+    dragonfly_job = if settings[:size] && settings[:crop]
+                      picture.crop(settings[:size], false, false, settings[:upsample] || false)
+                    else
+                      picture.resize(settings[:size])
+                    end
+
+    dragonfly_job = dragonfly_job.encode(settings[:format], options.compact.join(' '))
+    Alchemy::PictureVersion.from_cache(dragonfly_job)
+  end
+
+  ##
+  # Looks up the remote url for a Alchemy::PictureVersion, only works for S3DataStore.
+  #
+  # @param version_picture [Alchemy::PictureVersion]
+  def version_url(version)
+    if Dragonfly.app(:alchemy_pictures).datastore.is_a?(Dragonfly::S3DataStore)
```

Comment:
Is it intentional (and useful) that this returns `nil` if not using S3?

Reply:
the only real alternative currently in use would be a file store which doesn't provide URLs and would have to go through the pictures controller still.
So returning nil isn't ideal, but I don't really consider it worth it to support the pictures controller just for local deployments. I don't want to remove it just yet with this PR, but I think that is something simple enough that can happen as a follow up.